Add order status management and progression endpoints

barankonrad opened the pull request:
### Summary

This pull request introduces endpoints to handle order status progression through various stages (pending, confirmed, in-progress, ready). Additionally, it includes necessary validation, security updates, and tests to support these transitions.

### Details

- Created DTOs for orders and order items, with mappers.  
- Implemented validation for order status transitions.  
- Added endpoints to manage progression across statuses.  
- Included unit tests for status transitions and permission checks.  
- Refactored `OrderService` for better maintainability.  
- Updated security configurations to accommodate the new endpoints.

barankonrad committed:
Implement order status progression management

Add endpoints to move orders through different status stages (pending, confirmed, in-progress, ready)
- Create DTOs for order and order items with corresponding mappers
- Implement order status transition validation
- Add tests to verify status transitions and permission controls
- Refactor OrderService implementation
- Update security configuration to handle new endpoints

Piotr-Lenarczyk commented on backend/src/main/java/org/example/restaurantmanagementapplication/rest/OrderController.java:
```diff
@@ -1,65 +1,105 @@
+  }
+
+
+  @PostMapping("{id}/pending")
+  public ResponseEntity<OrderDto> setPending(@PathVariable int id) {
+    return updateStatus(id, OrderStatus.PENDING);
+  }
+
+  @PostMapping("{id}/confirmed")
+  public ResponseEntity<OrderDto> setConfirmed(@PathVariable int id) {
+    return updateStatus(id, OrderStatus.CONFIRMED);
+  }
+
+  @PostMapping("{id}/in_progress")
+  public ResponseEntity<OrderDto> setInProgress(@PathVariable int id) {
+    return updateStatus(id, OrderStatus.IN_PROGRESS);
+  }
+
+  @PostMapping("{id}/ready")
+  public ResponseEntity<OrderDto> setReady(@PathVariable int id) {
+    return updateStatus(id, OrderStatus.READY);
+  }
```
A general approach regarding API endpoints for state transitions is to use action names. That way, we do not expose internal logic to the client. I would propose:
```suggestion
  @PostMapping("{id}/mark-as-pending")
  public ResponseEntity<OrderDto> setPending(@PathVariable int id) {
    return updateStatus(id, OrderStatus.PENDING);
  }

  @PostMapping("{id}/confirm")
  public ResponseEntity<OrderDto> setConfirmed(@PathVariable int id) {
    return updateStatus(id, OrderStatus.CONFIRMED);
  }

  @PostMapping("{id}/start-preparation")
  public ResponseEntity<OrderDto> setInProgress(@PathVariable int id) {
    return updateStatus(id, OrderStatus.IN_PROGRESS);
  }

  @PostMapping("{id}/mark-as-ready")
  public ResponseEntity<OrderDto> setReady(@PathVariable int id) {
    return updateStatus(id, OrderStatus.READY);
  }
```
or something similar in the case of `setInProgress()` method.
Aside from that, all good! I like the parametrized tests :)

barankonrad replied:
Applied

barankonrad committed:
Refactor order endpoint URLs for improved clarity

Updated order management endpoint paths to use more descriptive and consistent naming conventions. Adjusted related security configurations and test cases accordingly to match the new endpoint structure.
